### PR TITLE
PIM-7570: Adds 'Locale specific' fields in attribute versioning

### PR DESCRIPTION
**Pull request by pierallard**
No description needed ?

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added legacy Behats               | y
| Added acceptance tests            | -
| Added integration tests           | -
| Changelog updated                 | y
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

### CODE REVIEW

File: .ci/Jenkinsfile [hunk 1]
```diff
@@ -41,7 +41,7 @@ stage("Build") {
                 withBuildNode({
                     checkout scm
                     container("php") {
-                        sh "php -d memory_limit=4G /usr/local/bin/composer update --ansi --optimize-autoloader --no-interaction --no-progress --prefer-dist --no-suggest"
+                        sh "php -d memory_limit=-1 /usr/local/bin/composer update --ansi --optimize-autoloader --no-interaction --no-progress --prefer-dist --no-suggest"
```
Comment (skeleton) [inline .ci/Jenkinsfile]: :allezla:

File: features/versioning/ensure_versioning_on_attribute.feature [hunk 2]
```diff
@@ -18,3 +18,14 @@ Feature: Ensure versioning on attribute
     And I press the "Save" button
     Then I visit the "History" tab
     Then there should be 2 update
+
+  @javascript @jira https://akeneo.atlassian.net/browse/PIM-7570
+  Scenario: Successfully version the locale_specific property
```
Comment (skeleton) [inline features/versioning/ensure_versioning_on_attribute.feature]: no way to add an integation test?

File: .ci/Jenkinsfile (hunk 1 above)
Comment (pierallard) [reply .ci/Jenkinsfile]: Cloud team ask me to do this `¯\_(ツ)_/¯`
:yolo:

File: src/Pim/Bundle/VersioningBundle/UpdateGuesser/AttributeLocaleUpdateGuesser.php [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+/**
+ * This class will guess if there are changes into the 'availableLocales' attribute field.
+ * If guessed, it will version the 'availableLocales' and the 'locale_specific' fields.
+ *
+ * @author    Pierre Allard <pierre.allard@akeneo.com>
+ * @copyright 2018 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class AttributeLocaleUpdateGuesser implements UpdateGuesserInterface
```
Comment (skeleton) [inline src/Pim/Bundle/VersioningBundle/UpdateGuesser/AttributeLocaleUpdateGuesser.php]: I'm not a big fan of the naming, I don't understand what it does :(

Comment (pierallard) [reply src/Pim/Bundle/VersioningBundle/UpdateGuesser/AttributeLocaleUpdateGuesser.php]: @skeleton What do you think of "LocaleSpecificAttributeUpdateGuesser" ? I put a big comment on the header to help understand.

**Pull request merged**